packages/shared/src/components/cards/squad/UnfeaturedSquadGrid.tsx

```diff
@@ -10,39 +10,46 @@ import {
 } from '../../typography/Typography';
 import { largeNumberFormat } from '../../../lib';
 import { Separator } from '../common';
-import { SquadImage } from './common/SquadImage';
-import { SquadJoinButtonWrapper } from './common/SquadJoinButton';
 import { UnFeaturedSquadCardProps } from './common/types';
+import { Origin } from '../../../lib/log';
+import { SquadJoinButton } from '../../squads/SquadJoinButton';
+import { ButtonVariant } from '../../buttons/common';
+import { Image, ImageType } from '../../image/Image';
 
 export const UnfeaturedSquadGrid = ({
   source,
-  title,
-  subtitle,
-  icon,
-  action,
-  description,
+  className,
 }: UnFeaturedSquadCardProps): ReactElement => {
+  const router = useRouter();
+  const title = source.name;
+
   return (
-    <Card className={classNames('min-w-[19rem] overflow-hidden border-0 p-4')}>
+    <Card
+      className={classNames(
+        'w-[19rem] overflow-hidden border-0 p-4',
+        className,
+      )}
+    >
       <CardLink
         href={source.permalink}
         rel="noopener"
         title={source.description}
       />
       <div className="mb-3 flex items-center justify-between">
-        <SquadImage
-          image={source?.image}
-          icon={icon}
-          title={title}
-          size="size-16"
+        <Image
+          src={source.image}
+          alt={`${title} source`}
+          className="size-16 rounded-full"
+          type={ImageType.Squad}
         />
-        <SquadJoinButtonWrapper
-          action={action}
-          source={source}
-          variant={ButtonVariant.Float}
-          className={{
-            squadJoinButton: '!btn-tertiaryFloat',
-          }}
+        <SquadJoinButton
+          className={{ button: 'z-0' }}
+          squad={source}
+          origin={Origin.SquadDirectory}
+          onSuccess={() => router.push(source?.permalink)}
+          data-testid="squad-action"
+          showViewSquad
```

Comment:
why is this prop here, can't we have a join button on this card?

Reply:
Still think it's misleading as a prop. From my understanding when I just see this file and not the component inside, what I get is that it will always show view only...

See my comment here: https://github.com/dailydotdev/apps/pull/3521#discussion_r1761122191

maybe that idea is not the best future proof as well but shows the current state. If u have something better in your mind please share 🙏

Reply:
If it is a matter of misunderstanding with the prop, we can always rename the variable.